elements/lisk-p2p/src/p2p_types.ts

```diff
@@ -53,55 +63,29 @@ export interface P2PInternalState {
 	readonly productivity?: number;
 	readonly reputation?: number;
 	readonly connectionKind?: ConnectionKind;
-	readonly advertiseAddress: boolean;
 }
 
-export interface P2PPeerInfo {
+export interface P2PBasicPeerInfo {
 	// String to uniquely identify each peer
-	readonly peerId: string;
+	readonly id: string;
 	readonly ipAddress: string;
-	readonly wsPort: number;
-	readonly sharedState?: P2PSharedState;
-	readonly internalState?: P2PInternalState;
 }
 
-export interface P2PPeersCount {
-	readonly outboundCount: number;
-	readonly inboundCount: number;
-}
-
-// P2PPeerInfo and P2PNodeInfo are related.
-// P2PNodeInfo is the outbound info from our node.
-export interface P2PNodeInfo extends P2PSharedState {
-	readonly protocolVersion: string;
-	readonly os: string;
-	readonly nethash: string;
-	readonly wsPort: number;
-	readonly advertiseAddress: boolean;
-	readonly nonce: string;
-}
-
-// This is a representation of the inbound peer object according to the current protocol.
-// TODO later: Switch to LIP protocol format.
-// TODO: Include peerId as field
-export interface ProtocolPeerInfo {
-	// To support the existing protocol
-	readonly ipAddress: string;
-	readonly wsPort: number;
-	// tslint:disable-next-line: no-mixed-interface
-	readonly [key: string]: unknown;
+export interface P2PPeerInfo extends P2PBasicPeerInfo {
+	readonly sharedState: P2PSharedState;
+	readonly internalState?: P2PInternalState;
 }
 
 export interface P2PConfig {
-	readonly blacklistedPeers?: ReadonlyArray<ProtocolPeerInfo>;
-	readonly seedPeers?: ReadonlyArray<ProtocolPeerInfo>;
-	readonly fixedPeers?: ReadonlyArray<ProtocolPeerInfo>;
-	readonly whitelistedPeers?: ReadonlyArray<ProtocolPeerInfo>;
-	readonly previousPeers?: ReadonlyArray<ProtocolPeerInfo>;
+	readonly sharedState: P2PSharedState;
```

Comment:
IMO, `nodeInfo` is still sounds more clear than `sharedState`. Having a distinction between nodeInfo and peerInfo is important because in future if we will have more fields like `advertiseAddress` then it will be hard to incorporate, maybe now they seem almost the same but still they are different

Reply:
What is `nodeInfo` for us, we share it to the rest of peers becoming `sharedState` when they receive it. I don't think there is a strong reasoning to keep this distinction. If more fields appear, `sharedState` will be the place to hold this information locally, the structure to share it and also to receive it. It just looks simple.